app/controllers/application_controller.rb

```diff
@@ -34,13 +34,17 @@ def adequate_scopes?
     required_scopes.all? { |scope| current_scopes.include?(scope) }
   end
 
+  def request_scopes
+    required_scopes | current_scopes
+  end
```

Comment:
Not sure yet, because we don't know what permissions the user already has because we don't know who the user is signing in before hand.